### PR TITLE
Redesign the channel choosing workflow

### PR DESCRIPTION
**Pull request by mixonic**
References https://github.com/brave-intl/publishers/issues/632, specifically part 1 where the UI for choosing a channel is laid out.

A few things tech-level things are included in this PR that set it up for a good implementation:

* Fix code whitespace in `omniauth_callbacks_controller.rb`
* Add a new feature to modals, `modal.js`. They will add a class that includes the identifier to the `.md-container` wrapping element. See the code for documentation.
* Rubyize/Underscore `authorize_email` `authorize_name`
* Refactor `choose_new_channel_type.html.slim` to use bootstrap 4.

So on the actual feature front:

* Update the style of the `choose_new_channel_type` page. This becomes the no-JS fallback flow.
* Add a modal version from the `publishers/home` page that re-uses the same DOM. Because the home page has not been converted to bootstrap 4, there is a lot of CSS added just to make that re-used DOM work well in the legacy styles. Once the `publishers/home` page is converted to bootstrap 4 (not in this PR) the page should mostly Just Work (tm).

<details>
  <summary>Click to expand for screenshots</summary>

<b>A) 960px wide screen, with JS</b>
localhost_3000_publishers_home_twitch 1 1

**B) Pixel2, with JS**
localhost_3000_publishers_home_twitch 1 pixel 2 1

**C) 960px wide screen, no JS**
localhost_3000_publishers_choose_new_channel_type_twitch 1
**D) Pixel2, no JS**
localhost_3000_publishers_choose_new_channel_type_twitch 1 pixel 2
</details>
<br>

By default, these screens do **not** have Twitch included. In order to enable Twitch, you need to add `twitch=1` to the URL and reload the page. For example use these links in development:

* https://localhost:3000/publishers/home?twitch=1
* https://localhost:3000/publishers/choose_new_channel_type?twitch=1

### CODE REVIEW

File: app/models/twitch_channel_details.rb [hunk 1]
```diff
@@ -15,11 +15,11 @@ def channel_identifier
     "twitch#channel:#{name}"
   end
 
-  def authorizerEmail
+  def authorize_email
```
Comment (ayumi) [inline app/models/twitch_channel_details.rb]: typo

File: app/views/publishers/_choose_channel_type.html.slim [hunk 2]
```diff
@@ -0,0 +1,27 @@
+= link_to(new_site_channel_path, class: 'card block-link--card') do
```
Comment (ayumi) [inline app/views/publishers/_choose_channel_type.html.slim]: this link doesn't actually work for me. it dismisses the modal and prevents default

Comment (mixonic) [reply app/views/publishers/_choose_channel_type.html.slim]: Addressed. In https://github.com/brave-intl/publishers/commit/1c60fc7f8a877441ab48397f3c27cc5170906522#diff-fdc082bd84b5d4eea4238f264d49e385R91 the `preventDefault` call was moved to incorrectly be fired any time the user clicked anywhere that was not the background of the page. Instead, `preventDefault` and other closing logic should only be called if `js-confirm` or `js-deny` is clicked.

**Comment by mixonic** [comment]
Rebased, also added a test for clicking on the website in the chooser.

**Pull request merged**